app/models/post.rb

```diff
@@ -1,2 +1,3 @@
 class Post < ApplicationRecord
+  belongs_to :user
```

Comment:
defining this connection here is what forces the post table to contain user_id?

Reply:
'belongs_to :user' is also a command found in the migration file when creating the table. The migration command understands that there will be a table called 'users' (Don't think it checks if it exists) and adds the 'user_id' column to the post table.  The 'belongs_to :user' found in the model is where the ORM makes the association between the two tables. I'm note sure what this association achieves on its own, validations can be added to confirm a user exists on the users table before a post is stored (this is done in the model). A foreign key has been added, this could be performing a validation instead.

Reply:
Don't think it checks if it exists > it does. you need to migrate the users table before the posts one